app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
+                $this->Session->setFlash(__('Search was disabled. Please try again later.', true));
+            } else {
+                $this->Session->setFlash(__('An error occurred while performing this function.', true));
```

Comment:
It would be good if users would let us know when there's an error with Sphinx, so this text could be changed into:
`An error occurred. If the error persists, please <a href="{}">contact us</a>.`

Something similar to: https://github.com/Tatoeba/tatoeba2/blob/dev/app/views/sentences/search.ctp#L87